portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,12 +27,21 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private ArrayList<String> comments = new ArrayList<String>( 
-    Arrays.asList("Comment 1", "Comment 2", "Comment 3"));
+    Arrays.asList());
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String json = new Gson().toJson(this.comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
-}
\ No newline at end of file
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String newComment = request.getParameter("new-comment");
+    if (!newComment.isEmpty()){
+      this.comments.add(newComment);
+    }
```

Comment:
It is not a requirement for this assignment but this is a great place to learn about HTTP error responses =D

https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html#sendError(int)

Here is a list of status codes to send back (they are under Field Summary)
https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html

For this case, it seems like you don't want empty strings to be a valid comment. So in this case you would to send error with the BAD_REQUEST code.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

Reply:
Always glad to learn! Now a BAD_REQUEST error is sent if the user inputs an empty comment. Thanks for the links.